src/swap/core/getQuote.ts

```diff
@@ -0,0 +1,40 @@
+import { CDP_GETSWAPQUOTE } from '../../definitions/swap';
+import { sendRequest } from '../../queries/request';
+import type {
+  GetQuoteResponse,
+  GetQuoteParams,
+  GetQuoteAPIParams,
+  Quote,
+  SwapError,
+} from '../types';
+import { getParamsForToken } from './utils';
+
+/**
+ * Retrieves a quote for a swap from Token A to Token B.
+ */
+export async function getQuote(params: GetQuoteParams): Promise<GetQuoteResponse> {
+  // Default parameters
+  const defaultParams = {
+    amountReference: 'from',
+    amountInDecimals: false,
+  };
+
+  params = { ...defaultParams, ...params };
+
+  const apiParams = getParamsForToken(params);
```

Comment:
Generally, you don't want to mutate the function argument

```suggestion
  const apiParams = getParamsForToken({ ...defaultParams, ...params });
```